plugin/query_predicates.lua

```diff
@@ -1,19 +1,59 @@
 local query = vim.treesitter.query
 
+local predicates = {
+  ---@param nodes TSNode[]
+  ---@param pred string[]
+  ---@param any boolean
+  ---@return boolean
+  ['has-type'] = function(nodes, pred, any)
+    if not nodes or #nodes == 0 then
+      return true
+    end
+
+    local types = { unpack(pred, 3) }
+    for _, node in ipairs(nodes) do
+      local match = vim.list_contains(types, node:type())
+      if any and match then
+        return true
+      elseif not any and not match then
+        return false
+      end
+    end
+    return not any
+  end,
+}
+
 -- register custom predicates (overwrite existing; needed for CI)
 
----@param match (TSNode|nil)[]
+---@param match TSMatch
 ---@param pred string[]
 ---@return boolean|nil
 query.add_predicate('has-type?', function(match, _, _, pred)
-  local node = match[pred[2]]
-  if not node then
-    return true
+  local nodes = match[pred[2]]
+  if type(nodes) ~= 'table' then
+    -- Check for old versions of Nvim that map capture IDs to a single node
+    nodes = { nodes }
```

Comment:
This branch targets neovim 0.10+, and I believe according to neovim/neovim#24738, this guard is unnecessary

Reply:
The guard is there in case someone updates nvim-treesitter before updating neovim (assuming both this PR and https://github.com/neovim/neovim/pull/24738 are merged). But you're right, since this targets 0.10 and we're talking about nightly/unstable versions, we probably don't need a guard and we can just tell people to update neovim.